libs/designer/src/lib/core/utils/parameters/helper.ts

```diff
@@ -463,16 +461,23 @@ const convertStringToInputParameter = (
   };
 };
 
+export const toArrayViewModelSchema = (schema: any): { arrayType: ArrayType; itemSchema: any } => {
+  const itemSchema = parseArrayItemSchema(schema);
+  const arrayType = schema?.type === constants.SWAGGER.TYPE.OBJECT && schema.properties ? ArrayType.COMPLEX : ArrayType.SIMPLE;
+  return { arrayType, itemSchema };
```

Comment:
Shouldn't you be having uncasted value added here? lets make sure we always add it regardless of deserialization/authoring scenario.

Reply:
if the uncastedValue is unset, it defaults to initialValue. I can make it so that we set uncastedValue to undefined though